lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformResourceAnnotations.php

```diff
@@ -39,23 +40,54 @@ public function buildAnnotations(Table $table) : self
 
     public function getPaginationPropertiesAnnotation(Table $table) : array
     {
-        $annotation = $table->parseComment(CustomComment::API_PLATFORM_PAGINATION);
-        if($$annotation) {
-            return [];
+        $comment = $table->parseComment(CustomComment::API_PLATFORM_PAGINATION);
+        $settings = $this->getKeyValueSettingsFromComment($comment);
+        $properties = [];
+        if(isset($settings['enabled'])) {
+            $properties[] = '    "pagination_enabled"=' . ($settings['enabled'] === 'true' ? 'true' : 'false');
+        } else {
+            //User has put in a pagination comment but not been explicit so assume thay want pagination on.
+            $properties[] = '    "pagination_enabled"=true';
         }
-        $parts = explode(self::COMMENT_FIELD_DELIMITER, $annotation);
-        $rawSetting = strtolower(trim($parts[0]));
-        $setting = $rawSetting == 'on' || $rawSetting == 'off' ? $rawSetting : 'on';
-        if(count($parts) > 1 &&  filter_var(trim($parts[1]), FILTER_VALIDATE_INT)){
-            $itemsPerPage = (int)trim($parts[1]);
+
+        if(isset($settings['items']) &&  filter_var($settings['items'], FILTER_VALIDATE_INT)){
+            $itemsPerPage = (int)trim($settings['items']);
+            $properties[] = ' "pagination_items_per_page"=' . $itemsPerPage;
+        }
+
+        if(isset($settings['max_items']) &&  filter_var($settings['max_items'], FILTER_VALIDATE_INT)){
+            $itemsPerPage = (int)trim($settings['max_items']);
+            $properties[] = ' "maximum_items_per_page"=' . $itemsPerPage;
+        }
+
+        if(isset($settings['client_control'])) {
+            $properties[] = ' "pagination_client_enabled"=' . ($settings['client_control'] === 'true' ? 'true' : 'false');
+            $properties[] = ' "client_items_per_page"=' . ($settings['client_control'] === 'true' ? 'true' : 'false');
+        } else {
+            //User has put in a pagination comment but not been explicit so assume thay want pagination on.
```

Comment:
is this comment correct? Looks like a reasonable aspiration but implementation appears to be disable

Reply:
Good catch - copy and paste error. Will fix intent.